fossunited/api/emailing.py

```diff
@@ -0,0 +1,451 @@
+from typing import Literal
+
+import frappe
+
+from fossunited.api.chapter import check_if_chapter_member
+from fossunited.doctype_ids import EVENT
+
+EMAIL_GROUP_TYPES = Literal[
+    "Chapter Master",
+    "Event Participants",
+    "CFP Proposers",
+    "Accepted Proposers",
+    "Rejected Proposers",
+    "Other",
+]
+
+
+def create_email_group(
+    event_id: str,
+    type: EMAIL_GROUP_TYPES,
+):
+    """
+    Create an email group linked to the event
+
+    Args:
+        event: event id
+        session_user: session user id
+    """
+    _event = frappe.get_doc(EVENT, event_id)
+
+    # if not check_if_chapter_member(_event.chapter, session_user):
+    #     raise frappe.PermissionError("You are not authorised for this action.")
+
+    if frappe.db.exists(
+        "Email Group", {"event": event_id, "chapter": _event.chapter, "group_type": type}
+    ):
+        raise frappe.ValidationError("Email Group already exists for this event")
+
+    group_title = f"{type}-{_event.event_name}-{_event.chapter_name}-{event_id}"
+    _group_title = group_title[:140]
```

Comment:
this feels suboptimal. i guess what we're trying to do here is to create a unique title for the doc. and the `title` field is used elsewhere, which is why it needs to be meaningful, is that right?

If so, can we just remove the `{type}` and `{_event.chapter_name}` from the format because they are already stored on the `group` object. we can reconstruct the string from the data if necessary.

Can you demo a few examples here i.e. what does `group_title` look like?

Reply:
![image](https://github.com/user-attachments/assets/85b9113e-204e-4141-9a42-8b639f205abb)

maybe the best way to handle this would be to extend email group to have a custom auto naming similar to Newsletter doctype?

Reply:
That way we can have the IDs to be random hash

Reply:
> maybe the best way to handle this would be to extend email group to have a custom auto naming similar to Newsletter doctype?

i'm not sure.

also, what's wrong with `{type}-{event id}` like I had mentioned earlier? The `FOSS Meetup-BANGALORE` doesn't add any meaningful information to the id and even if there is a condition for the ID to be unique, because the event ID is unique, this should be fine. Right?

Reply:
this would work too, i was suggesting an alternative. Will be making changes as per suggestions